traversal/procedure/ProcedureVertex.java

```diff
@@ -167,7 +151,6 @@ public Thing asThing() {
 
         @Override
         public Forwardable<? extends ThingVertex, Order.Asc> iterator(GraphManager graphMgr, Traversal.Parameters parameters) {
-            assert isStartingVertex();
```

Comment:
This precondition was only satisfied accidentally. While it is true that GraphProcedure / GraphIterator should never call this on a non-starting vertex, CombinationProcedure also uses this class, and it does not assign any kind of ordering.